packages/contracts/libs/common/src/attestation.rs

```
@@ -0,0 +1,282 @@
+pub fn build_payload_bytes(env: &Env, payload: &AttestationPayload, nonce: u64) -> Bytes {
+    // source_id as 8 big-endian bytes of the Symbol's packed value. This SDK
+    // version gives Symbol neither Display nor to_xdr, and a symbol_short is
+    // already a packed u64, so encode those bits directly: fixed-width,
+    // distinct for distinct symbols, and reproducible by the off-chain signer.
+    let sym_raw: u64 = payload.source_id.to_val().get_payload();
+    let mut sym_buf = [0u8; 8];
+    for (i, slot) in sym_buf.iter_mut().enumerate() {
+        *slot = ((sym_raw >> (56 - 8 * i)) & 0xff) as u8;
+    }
+    let id_len_u32: u32 = 8;
+    let id_len_n = 8usize;
```

Clippy (1.97.1)

[warning] unused variable

(warning)